packages/salesforcedx-vscode-core/src/orgBrowser/metadataOutlineProvider.ts

```diff
@@ -20,37 +21,107 @@ export class MetadataOutlineProvider
     BrowserNode | undefined
   > = this.internalOnDidChangeTreeData.event;
 
-  constructor(defaultOrg?: string) {
+  constructor(defaultOrg: string | undefined) {
     this.defaultOrg = defaultOrg;
   }
 
-  public async refresh(defaultOrg?: string): Promise<void> {
-    if (!isNullOrUndefined(defaultOrg)) {
-      this.defaultOrg = defaultOrg;
+  public async onViewChange() {
+    const usernameOrAlias = await this.getDefaultUsernameOrAlias();
+    if (usernameOrAlias !== this.defaultOrg) {
+      this.internalOnDidChangeTreeData.fire();
     }
+    this.defaultOrg = usernameOrAlias;
+  }
+
+  public async refresh(): Promise<void> {
+    const usernameOrAlias = await this.getDefaultUsernameOrAlias();
+    this.defaultOrg = usernameOrAlias;
     this.internalOnDidChangeTreeData.fire();
   }
 
   public getTreeItem(element: BrowserNode): vscode.TreeItem {
     return element;
   }
 
-  public getChildren(element?: BrowserNode): Promise<BrowserNode[]> {
-    if (isNullOrUndefined(element)) {
-      if (!isNullOrUndefined(this.defaultOrg)) {
-        const org = new BrowserNode(this.defaultOrg, NodeType.Org);
-        return Promise.resolve([org]);
-      } else {
-        return Promise.resolve([]);
-      }
-    } else {
-      return Promise.resolve(element.children);
+  public async getChildren(element?: BrowserNode): Promise<BrowserNode[]> {
+    if (isEmpty(this.defaultOrg)) {
+      const emptyDefault = new BrowserNode(
+        nls.localize('missing_default_org'),
+        NodeType.EmptyNode
+      );
+      return Promise.resolve([emptyDefault]);
+    }
+
+    if (isEmpty(element)) {
+      const org = new BrowserNode(this.defaultOrg, NodeType.Org);
+      return Promise.resolve([org]);
     }
+
+    switch (element.type) {
+      case NodeType.Org:
+        element.children = await this.getTypes();
+        break;
+      case NodeType.MetadataType:
+        element.children = await this.getComponents(element);
+        break;
+    }
+    return Promise.resolve(element.children);
   }
 
-  public async getDefaultUsernameOrAlias() {
+  public async getTypes(): Promise<BrowserNode[]> {
```

Comment:
getTypes(), getComponents(), and getDefaultUsernameOrAlias() don't need to be public.

Reply:
A few of the methods need to be public so I can access them for testing.